File: backend/src/main/java/com/ironlogic/modules/program/domain/model/SessionTemplate.java
package com.ironlogic.modules.program.domain.model;

import java.time.LocalDateTime;

/**
 * SessionTemplate 的领域模型。
 *
 * <p>SessionTemplate 归属于某个 ProgramBlock，用于定义可复用的训练日模板，
 * 例如 Push Day、Leg Day。
 *
 * @param id SessionTemplate id
 * @param blockId 所属 Block id
 * @param name 模板名称
 * @param sequenceNo 在 Block 内的顺序号
 * @param triggerMode 触发模式
 * @param notes 可选备注
 * @param metadataJson 预留元数据 JSON
 * @param createdAt 创建时间
 * @param updatedAt 更新时间
 */
public record SessionTemplate(
        Long id,
        Long blockId,
        String name,
        Integer sequenceNo,
        String triggerMode,
        String notes,
        String metadataJson,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {
}

File: frontend/lib/features/program/domain/model/session_template.dart
/// SessionTemplate 领域模型。
///
/// 该类属于 `program/domain`，表示某个 Block 下的训练日模板。
/// 它仍然属于模板定义层，不包含实际训练结果。
class SessionTemplate {
  /// 创建 SessionTemplate 模型。
  const SessionTemplate({
    required this.id,
    required this.blockId,
    required this.name,
    required this.sequenceNo,
    this.triggerMode,
    this.notes,
    this.metadataJson,
    this.createdAt,
    this.updatedAt,
  });

  /// 模板 id。
  final int id;

  /// 所属 Block id。
  final int blockId;

  /// 模板名称。
  final String name;

  /// 顺序号。
  final int sequenceNo;

  /// 触发模式。
  final String? triggerMode;

  /// 备注。
  final String? notes;

  /// 元数据 JSON。
  final String? metadataJson;

  /// 创建时间。
  final DateTime? createdAt;

  /// 更新时间。
  final DateTime? updatedAt;
}
